Makefile.system

```diff
@@ -9,6 +9,11 @@ ifndef TOPDIR
 TOPDIR = .
 endif
 
+# If ARCH is not set, we use the host system's architecture.
+ifndef ARCH
+ARCH := $(shell uname -m)
```

Comment:
I hope this has no side effects when doing cross compilation. I have no setup to test this. If it has, of course we can use another variable here.

Reply:
My gut feeling is to use some local name that cannot ever conflict with the general ARCH setting here, but we will see how this goes.